libparsec/crates/protocol/schema/authenticated_cmds/invite_greeter_step.json5

```diff
@@ -0,0 +1,207 @@
+                "status": "attempt_not_joined"
+            },
+            {
+                // The attempt has been cancelled
+                "status": "attempt_cancelled"
+            },
+            {
+                // The submitted step is too advanced
+                // Every step before must have been completed by both peers
+                "status": "step_too_advanced"
+            },
+            {
+                // The submitted step somehow changed during polling
+                "status": "step_mismatch"
+            }
+        ],
+        "nested_types": [
+            {
+                // GreeterStep should be identical to the one in invite_claimer_step.json5
+                "name": "GreeterStep",
+                "discriminant_field": "step",
+                "variants": [
+                    {
+                        "name": "WaitPeer",
+                        "discriminant_value": "0",
+                        "fields": [
+                            {
+                                "name": "public_key",
+                                "type": "PublicKey"
+                            }
+                        ]
+                    },
+                    {
+                        "name": "GetHashedNonce",
+                        "discriminant_value": "1"
+                    },
+                    {
+                        "name": "SendNonce",
+                        "discriminant_value": "2",
+                        "fields": [
+                            {
+                                "name": "greeter_nonce",
+                                "type": "Bytes"
+                            }
+                        ]
+                    },
+                    {
+                        "name": "GetNonce",
+                        "discriminant_value": "3"
+                    },
+                    {
+                        "name": "WaitPeerTrust",
+                        "discriminant_value": "4"
+                    },
+                    {
+                        "name": "SignifyTrust",
+                        "discriminant_value": "5"
+                    },
+                    {
+                        "name": "GetPayload",
+                        "discriminant_value": "6"
+                    },
+                    {
+                        "name": "SendPayload",
+                        "discriminant_value": "7",
+                        "fields": [
+                            {
+                                "name": "greeter_payload",
+                                "type": "Bytes"
+                            }
+                        ]
+                    },
+                    {
+                        "name": "WaitPeerAcknowledgment",
+                        "discriminant_value": "8"
+                    }
+                ]
+            },
+            {
+                // ClaimerStep should be identical to the one in invite_claimer_step.json5
```

Comment:
Could it be defined in types instead of being a nested type?

Reply:
I agree that they should, but I'm not sure they can. So far, the `"variants"` key is only used in `"nested_types"`, so I don't think the macros support the definition an enum from the json schema (without using `nested_types`).

Another approach would be to use a nested type from another command in a command definition, but macros would need to be adapted as well. My thoughts were to duplicate those `ClaimerStep` and `GreeterStep` definition, add a comment about this, a refactor this by updating the macros later.

@touilleMan could you share your thoughts about this?